client/signup/steps/domains/index.jsx

```diff
@@ -111,11 +111,20 @@ module.exports = React.createClass( {
 			stepSectionName: this.props.stepSectionName
 		}, this.getThemeArgs() ), [], { domainItem } );
 
-		if ( domainsWithPlansOnlyTestEnabled && isPurchasingItem ) {
+		if ( domainsWithPlansOnlyTestEnabled && isPurchasingItem && abtest( 'freeTrialsInSignup' ) !== 'enabled' ) {
 			this.submitPlansStepWithPremium();
-		}
+			const plansIndex = this.props.steps.indexOf( 'plans' );
+			if ( plansIndex === this.props.steps.length - 1 || plansIndex === -1 ) {
+				// if plans is the last step or not in the flow, this'll finish the flow
+				this.props.goToNextStep();
+			} else {
+				// skip plans step
 
-		this.props.goToNextStep();
+				this.props.goToStep( this.props.steps[ plansIndex + 1 ] );
+			}
+		} else {
```

Comment:
seems like the nested if/else cases could be a little cleaner, but don't let that hold up this fix for a production issue. might want to rework it to be a little simpler when you have more time.